pkg/controller/external.go

```diff
@@ -162,25 +162,28 @@ func (e *external) Observe(ctx context.Context, mg xpresource.Managed) (managed.
 		return managed.ExternalObservation{}, errors.Wrap(err, "cannot get connection details")
 	}
 
+	lateInitedParams, err := tr.LateInitialize(res.State.GetAttributes())
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, "cannot late initialize parameters")
+	}
+	lateInitialized := annotationsUpdated || lateInitedParams
+	resourceReady := tr.GetCondition(xpv1.TypeReady).Equal(xpv1.Available())
```

Comment:
This is to make sure that if `unavailable -> available` transition is necessary, we will update the status hence no need to do any spec-related operations since they will be lost anyway. So `alreadyAvailable` or `markedAvailable` could be a less confusing than `resourceReady` because it doesn't represent the status in the cloud, it's only about whether the resource is marked as ready in etcd.